cross/gdbm/Makefile

```diff
@@ -16,4 +16,7 @@ GNU_CONFIGURE = 1
 CONFIGURE_ARGS  = --disable-static
 CONFIGURE_ARGS += --enable-libgdbm-compat
 
+# fix unknown type blksize_t for older compilers (https://github.com/Homebrew/linuxbrew-core/pull/6242)
+ADDITIONAL_CFLAGS = -D_GNU_SOURCE
```

Comment:
Perhaps you could check if DSM <= 5|6.x or similar to not affect DSM7 builds?  Unless all builds are affected?

Reply:
This will need some redesign in the *.mk files to allow TC_GCC version dependent definitions.
As this does not break things, I leave it for the moment...